### PR TITLE
Query and update animals

### PR DESCRIPTION
**Pull request by braincee**
_**In this pull request:**_

- I added a column species of type string to your animal's table and Modified my `schema.sql` file using ALTER command.
- I inserted the following data into the `data.sql` file:
    - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has 
       never tried to escape.
   - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has 
      tried to escape 2 times.
   - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he 
     has tried to escape 3 times.
   - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried 
     to escape once.
   - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried 
      to escape 7 times.
  -  Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried 
     to escape 3 times.
  -  Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to 
     escape 4 times.

Screenshot (135)

- I added a transaction update for the animals' table by setting the species to unspecified and rollback.

Screenshot (136)

- I updated the animals' table by setting the species column to digimon for all animals that have a name ending in mon.
- I updated the animals' table by setting the species column to pokemon for all animals that don't have species already set.
 - I committed the transaction and verified the changes made persisted after the commit.
 
Screenshot (137)

- In a delete transaction, all the animals in the table were deleted. Then I rolled back the transaction. After the rollback, I verified if all the transactions exists.

Screenshot (138)

- In the next transaction I deleted all animals born on January 1st, 2022, then created a savepoint for the transaction.

Screenshot (139)

- I updated the animals' weight to be their weight multiplied by -1 then rolled back the savepoint. 

Screenshot (140)

- I updated all animals' weights that are negative to be their weight multiplied by -1 and then committed the transaction.

Screenshot (141)

- I wrote queries to answer the following questions
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  
 
Screenshot (143)

   - What is the minimum and maximum weight of each type of animal?
   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Screenshot (144)

## Changes Requirement screenshots.

- I have made verification before and after ROLLBACKS changes for species.

Screenshot (148)

- I have made verifications to changes before and after commits.

Screenshot (149)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -15,3 +15,44 @@ SELECT * FROM animals WHERE neutered = True;
 SELECT * FROM animals WHERE name != 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+--Begin animal transaction
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
```
Comment (DeliceLydia) [inline queries.sql:22]: - [ ] I suggest that you verify that change was made and if it is persisting:
```
` SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
`
```

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,44 @@ SELECT * FROM animals WHERE neutered = True;
...
+ROLLBACK;
+
+-- another animal transaction
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (DeliceLydia) [inline queries.sql]: - [ ] Same here before committing please verify:
```
 SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @braincee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by EricMbouwe — APPROVED** [review]
Hi @braincee, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good PR
- Linters ok
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by braincee** [comment]
Thanks very much, @EricMbouwe  and @DeliceLydia 

thank-you

**Pull request merged**